### PR TITLE
added Inventory Resource, Mining update and Shop Sell

### PR DESCRIPTION
**Pull request by ItsMeAdil**

### CODE REVIEW

File: bot/interactions/commands/economy/economyMine.ts [hunk 1]
```diff
@@ -239,71 +234,79 @@ export const mine = {
...
+    const resourceQuantity = await getResourceQuantity(
+      selectedPickaxe,
+      randomizedResources,
+    );
+    const decrement = decrementDurability[randomizedResources];
 
-    const clanBonus = Math.round(reward * clanBonusMultiplier);
-    const totalReward = reward + clanBonus;
+    /**Checks if the ResourceData has an ID which would indicate that it's a resource, Else it would normally send it to scenario */
+    if (resourceData.id !== "" && resourceData.id !== "") {
```
Comment (TheUndo) [inline bot/interactions/commands/economy/economyMine.ts:247]: ?
```suggestion
    if (resourceData.id) {
```

File: bot/interactions/commands/economy/shop/economyFunction.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+import { addCurrency } from "!/bot/utils/addCurrency";
+import { formatItem } from "!/bot/utils/formatItem";
+import { formatNumber } from "!/bot/utils/formatNumber";
+import { prisma } from "!/core/db/prisma";
+import type { Guild, User } from "discord.js";
+import { sprintf } from "sprintf-js";
+import { ItemType } from "../lib/shopConfig";
+import { formatSellResourceItems } from "./economyShop";
+
+type dataType = {
```
Comment (TheUndo) [inline bot/interactions/commands/economy/shop/economyFunction.ts:11]:
```suggestion
type DataType = {
```

Comment (TheUndo) [reply bot/interactions/commands/economy/shop/economyFunction.ts:11]: maybe rename to something less generic?

File: bot/interactions/commands/economy/shop/economyFunction.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        },
+      },
+    },
+
+    select: {
+      level: true,
+    },
+  });
+
+  console.log(userClan?.level);
```
Comment (TheUndo) [inline bot/interactions/commands/economy/shop/economyFunction.ts:64]:
```suggestion
```

File: bot/interactions/commands/economy/shop/economyFunction.ts [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+type data = {
+  user: User;
+  guild: Guild;
+};
+
+type createSoldEmbedType = {
+  data: data;
+  inventory: inventoryType[];
+  price: priceType;
+};
```
Comment (TheUndo) [inline bot/interactions/commands/economy/shop/economyFunction.ts:117]: all of these types should be pascal case

File: bot/interactions/commands/economy/shop/economyShop.ts [hunk 5]
```diff
@@ -151,6 +159,232 @@ const formatBuyToolItems = async (user: User, guild: Guild) => {
   return { embed, component: [firstRow] };
 };
 
-// const formatSellResourceItems = async (user: User, guild: Guild) => {
+type fieldValue = {
```
Comment (TheUndo) [inline bot/interactions/commands/economy/shop/economyShop.ts:162]:
```suggestion
type FieldValue = {
```

File: bot/interactions/commands/inventory/inventoryInteraction.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+} from "./inventory";
+
+export async function inventoryView(
+  interactionContext: InteractionContext,
+  interaction: AnyInteraction,
+) {
+  if (!interaction.isButton()) {
+    return void (await interaction.reply({
+      ephemeral: true,
+      content: "This interaction is an button",
```
Comment (TheUndo) [inline bot/interactions/commands/inventory/inventoryInteraction.ts:18]:
```suggestion
      content: "This interaction is a button",
```

**Review by TheUndo — CHANGES_REQUESTED** [review]
The logic looks ok but what I'm wondering is, is the sell and inventory only for "resources"? When we have non-resource items, is your intention to use the same system as for resources or create a parallel system? My concern is the shop/inventory/sell is closely tied to mine when that is only one aspect of the shop system

**Comment by ItsMeAdil** [comment]
Im planning on doing like what i did to inventory where we create a whole new embed page for different items other than resources.

Example:
- for resource
```ts
export async function economyShopResourceSellMenu(
  interactionContext: InteractionContext,
  interaction: AnyInteraction,
) {}
```

- pet
```ts
export async function economyShopPetSellMenu(
  interactionContext: InteractionContext,
  interaction: AnyInteraction,
) {}
```

**Review by TheUndo — APPROVED** [review]

**Pull request merged**